util.go

```diff
@@ -1,12 +1,62 @@
 package amt
 
-import "github.com/filecoin-project/go-amt-ipld/v3/internal"
+import (
+	"math"
+	"math/bits"
+)
 
-func nodesForHeight(height int) uint64 {
-	heightLogTwo := uint64(internal.WidthBits * height)
-	if heightLogTwo >= 64 {
-		// Should never happen. Max height is checked at all entry points.
-		panic("height overflow")
+// Exponentiation by squaring, saturating at max uint64.
+//
+// This saturates at max uint64 because:
+// 1. It's "correct" in that that's the maximum number of nodes we can have.
+// 2. It produces the correct results.
```

Comment:
Please check this (happy to discuss but it would be nice to get independent confirmation).